File: src/main/java/no/jstien/bikeapi/tsdb/read/Filter.java
package no.jstien.bikeapi.tsdb.read;

import java.util.Arrays;
import java.util.stream.Collectors;

public class Filter {
    private String type;
    private String tagk;
    private String filter;
    private boolean groupBy;

    public static Filter literalOrFilter(String tagk, boolean groupBy, int ...tagValues) {
        String[] objs = new String[tagValues.length];
        for (int i=0; i<tagValues.length; i++) {
            objs[i] = String.valueOf(tagValues[i]);
        }

        return literalOrFilter(tagk, groupBy, objs);
    }

    public static Filter literalOrFilter(String tagk, boolean groupBy, String ...tagValues) {
        Filter filter = new Filter();
        filter.type = "literal_or";
        filter.tagk = tagk;
        filter.groupBy = groupBy;
        filter.setFilter(tagValues);
        return filter;
    }

    public void setFilter(String ...tagValues) {
        filter = Arrays.stream(tagValues)
                .map(Object::toString)
                .collect(Collectors.joining("|"));
    }
}

File: src/main/java/no/jstien/bikeapi/tsdb/read/Filter.kt
package no.jstien.bikeapi.tsdb.read

import java.util.Arrays
import java.util.stream.Collectors

class Filter {
    private var type: String? = null
    private var tagk: String? = null
    private var filter: String? = null
    private var groupBy: Boolean = false

    fun setFilter(vararg tagValues: String) {
        filter = Arrays.stream(tagValues)
                .map{ i -> i.toString() }
                .collect(Collectors.joining("|"))
    }

    companion object {

        fun literalOrFilter(tagk: String, groupBy: Boolean, vararg tagValues: Int): Filter {
            val objs = Array(tagValues.size, { i -> tagValues[i].toString() })
            return literalOrFilter(tagk, groupBy, *objs)
        }

        fun literalOrFilter(tagk: String, groupBy: Boolean, vararg tagValues: String): Filter {
            val filter = Filter()
            filter.type = "literal_or"
            filter.tagk = tagk
            filter.groupBy = groupBy
            filter.setFilter(*tagValues)
            return filter
        }
    }
}
